src/WebJobs.Extensions.DurableTask/RetryOptionsConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class RetryOptionsConverter : JsonConverter
+    {
+        private const string BackoffCoefficientField = "backoffCoefficient";
+        private const string FirstRetryIntervalField = "firstRetryIntervalInMilliseconds";
+        private const string MaxAttemptsField = "maxNumberOfAttempts";
+        private const string MaxRetryIntervalField = "maxRetryIntervalInMilliseconds";
+        private const string RetryTimeoutField = "retryTimeoutInMilliseconds";
+
+        public override bool CanWrite => false;
+
+        public override bool CanConvert(Type objectType) => objectType.IsAssignableFrom(typeof(RetryOptions));
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            JObject obj = JObject.Load(reader);
+
+            if (!this.FieldExists(FirstRetryIntervalField, obj))
+            {
+                throw new ArgumentException("Invalid JSON. Must contain field firstRetryIntervalInMilliseconds", "reader");
+            }
+
+            if (!this.FieldExists(MaxAttemptsField, obj))
+            {
+                throw new ArgumentException("Invalid JSON. Must contain field maxNumberOfAttempts", "reader");
+            }
+
+            var firstRetryIntervalInMilliseconds = obj[FirstRetryIntervalField].Value<int>();
+            var maxNumberOfAttempts = obj[MaxAttemptsField].Value<int>();
```

Comment:
It just occurred to me that you're doing double reads on these fields. Rather than having a `FieldExists` method, it would be more efficient to just read once using TryGetValue, like I suggested in my other comment (except you would need to keep the value instead of discarding it).